src/plugins/intel_gna/tests/functional/pass_tests/concat_memory_param.cpp

```diff
@@ -0,0 +1,134 @@
+            blob->allocate();
+
+            auto* raw_blob_data_ptr = blob->buffer().as<float*>();
+            std::vector<float> values = CommonTestUtils::generate_float_numbers(blob->size(), -2.f, 2.f);
+            for (size_t i = 0; i < blob->size(); i++) {
+                raw_blob_data_ptr[i] = values[i];
+            }
+            return blob;
+        }
+
+        void SetUp() override {
+            InferenceEngine::Precision net_prc;
+            std::vector<size_t> input_shape;
+            std::tie(net_prc, targetDevice, configuration, input_shape) = this->GetParam();
+
+            auto ng_prc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(net_prc);
+
+            size_t in_total_dims_size = ov::shape_size(input_shape);
+            auto params = ngraph::builder::makeParams(ng_prc, {{1, in_total_dims_size}});
+            auto reshape_pattern = std::make_shared<ngraph::opset9::Constant>(ov::element::Type_t::i64, ov::Shape{2}, input_shape);
+            auto reshape = std::make_shared<ngraph::opset9::Reshape>(params[0], reshape_pattern, false);
+
+            ov::op::util::VariableInfo vi{};
+            vi.data_shape = ov::PartialShape(input_shape);
+            vi.variable_id = "test_variable";
+            vi.data_type = ov::element::Type_t::f32;
+            const auto var = std::make_shared<ov::op::util::Variable>(vi);
+            std::vector<float> initial_state = CommonTestUtils::generate_float_numbers(in_total_dims_size, -3.f, 3.f);
+            auto initial_state_node = ngraph::builder::makeConstant(ov::element::Type_t::f32, input_shape, initial_state);
+            auto readValue = std::make_shared<ngraph::opset9::ReadValue>(initial_state_node, var);
+
+            const int axis = 1;
+            ov::OutputVector to_concat{readValue, reshape};
+            auto concat = ngraph::builder::makeConcat(to_concat, axis);
+
+            const auto concat_shape = concat->get_output_shape(0);
+            const auto concat_shape_size = ov::shape_size(concat_shape);
+
+            auto etlwise_data = CommonTestUtils::generate_float_numbers(concat_shape_size, -1.f, 1.f);
+            auto etlwise_node = ngraph::builder::makeConstant(ov::element::Type_t::f32, concat_shape, etlwise_data);
+            auto etlwise_result_node = std::make_shared<ngraph::opset9::Multiply>(concat, etlwise_node);
+
+            ov::ResultVector results{std::make_shared<ngraph::opset9::Result>(etlwise_result_node)};
+            auto split_node = ngraph::builder::makeSplit(concat, ng_prc, 2, axis);
+
+            auto assign_node = std::make_shared<ngraph::opset9::Assign>(split_node->output(1), var);
+            ngraph::SinkVector sinks{assign_node};
+            function = std::make_shared<ov::Model>(results, sinks, params);
+        }
+};
+
+TEST_P(ConcatMemoryTest, CompareWithRefImpl) {
+    Run();
+};
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+    InferenceEngine::Precision::FP32,
```

Comment:
We could add FP16 as in other tests.
Although I've never seen any FAILURE affecting only one of this two (FP16, FP32) so I guess coverage will not increase much.